src/backends/gpu/fvm.hpp

```diff
@@ -76,20 +85,41 @@ struct backend {
         nest::mc::gpu::threshold_watcher<value_type, size_type>;
 
     // perform min/max reductions on 'array' type
-    static std::pair<value_type, value_type> minmax_value(const array& v) {
-        // TODO: replace with CUDA kernel
+    template <typename V>
+    static std::pair<V, V> minmax_value(const memory::device_vector<V>& v) {
+        // TODO: consider/test replacement with CUDA kernel (or generic reduction kernel).
         auto v_copy = memory::on_host(v);
         return util::minmax_value(v_copy);
     }
 
+    // perform element-wise comparison on 'array' type against `t_test`.
+    template <typename V>
+    static bool any_time_before(const memory::device_vector<V>& t, V t_test) {
+        // Note: benchmarking (on a P100) indicates that using the gpu::any_time_before
+        // function is slower than the copy, unless we're running over ten thousands of
+        // cells per cell group.
+        //
+        // Commenting out for now, but consider a size-dependent test or adaptive choice.
+
+        // return gpu::any_time_before(t.size(), t.data(), t_test);
+
+        cudaDeviceSynchronize();
```

Comment:
It is not necessary; this is a relic of the effort to track down race condition issues. Will remove.